c/driver_manager/adbc_driver_manager.cc

```diff
@@ -781,12 +810,17 @@ AdbcStatusCode AdbcLoadDriverFromInitFunc(AdbcDriverInitFunc init_func, int vers
     return ADBC_STATUS_INTERNAL;                                               \
   }
 
-  auto result = init_func(version, raw_driver, error);
+  AdbcStatusCode result = ADBC_STATUS_NOT_IMPLEMENTED;
+  for (const int try_version : kSupportedVersions) {
+    if (try_version > version) continue;
```

Comment:
I'm trying to understand the logic. Does it mean that, if the user passes a future `ADBC_VERSION_1_2_0`, we try calling `init_func` with `ADBC_VERSION_1_1_0` and then with `ADBC_VERSION_1_0_0`?

Reply:
Perhaps you should explain the intent in a comment.

Reply:
Right, so as mentioned above this should check for a supported version first. I've clarified in the source: the intent is that if they pass 1_1_0, we'll try to pass that to the underlying driver, then try 1_0_0 with the driver. If they pass 1_0_0 we'll only try 1_0_0.